src/shared/utils/index.ts

```diff
@@ -28,6 +28,24 @@ export async function fetchValidatorList(url: string): Promise<UNLBlob> {
   }
 }
 
+/**
+ * Get the network entry node's url for a validator.
+ *
+ * @param key - The public key of the validator.
+ * @returns A promise that resolves to the network entry url string.
+ */
+async function getNetworksEntryUrl(key: string): Promise<string | null> {
```

Comment:
some endpoints search by signing key and others by master key.  We should accept both or choose which one to consistently use.